enterprise/server/remote_execution/container/container.go

```diff
@@ -84,7 +84,7 @@ type Provider interface {
 // It's not strictly necessary to look at all of these params in order to create
 // a working container implementation, but they might be useful.
 type Init struct {
-	// WorkDir is the working directory for the initially assigned task.
+	// WorkDir is the working directory for the *initially* assigned task.
 	WorkDir string
```

Comment:
Can we eliminate this parameter in favor ofTask.WorkDir?

Reply:
In general - could we remove these fields that apply "to the initially assigned task" and rely on the Task struct passed into each specific function? It feels confusing and error prone to persist this "initial state"

Reply:
Yeah, I meant to add a TODO here to remove `Init` in favor of having `Run()` / `Create()` just look at the task struct - Done. At the end of the day, `provider.New()` should just be returning an empty struct, and the lifecycle methods should actually be responsible for initialization.